server/zanata-war/src/test/java/org/zanata/ZanataRestTest.java

```diff
@@ -79,12 +84,11 @@ public final void prepareRestEasyFramework() {
                     providerInstance);
         }
 
-        InMemoryClientExecutor executor =
-                new InMemoryClientExecutor(dispatcher);
```

Comment:
You're right, it's hard to find anything, but I think this approach might work as an alternative to InMemoryClientExecutor and friends, with mock support and so on: https://paluch.biz/blog/135-how-to-test-a-rest-api-with-junit.html

It should be a bit more realistic too, since it will make real HTTP requests.

Anyway, I've just got TMXDummyRestTest to work with the current code (yours), at least for a dead simple case (I shouldn't have picked such a complex service to test). 

It should be just enough to serve as an example for other REST tests, and also to let us try out alternatives to InMemoryClientExecutor - if the test is green, everything must be hooked up okay.

I'll put it into a separate PR, to be merged after this one. Please don't amend any commits or use squash merge!